go/integration_test/processor/calls_test.go

```diff
@@ -176,37 +171,274 @@ func TestProcessor_DifferentCallTypesAccessStorage(t *testing.T) {
+					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 12), sender0[:]...)) {
+						t.Errorf("%s did not set the correct sender, got unexpected output: %v", callName, result.Output)
+					}
+				} else {
+					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 12), receiver0[:]...)) {
+						t.Errorf("%s did not set the correct sender, got unexpected output: %v", callName, result.Output)
 					}
 				}
 			})
 		}
 	}
 }
 
+func TestProcessor_RecursiveCallsAfterAStaticCallAreStatic(t *testing.T) {
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue // todo implement different call types
+		}
+		calls := callTypesAndProperties()
+		for callName, call := range calls {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+
+				// Test structure:
+				// sender0 --CALL--> receiver0 --STATICCALL--> receiver1
+				// --test.CALL--> receiver2 -> SSTORE
+				// after the first static calls all following calls have to be static,
+				// therefore the SSTORE should revert. The success boolean of the test.CALL
+				// is returned as output.
+
+				sender0 := tosca.Address{1}
+				receiver0 := tosca.Address{2}
+				receiver1 := tosca.Address{3}
+				receiver2 := tosca.Address{4}
+
+				code0 := pushCallArguments(calls["staticCall"], sufficientGas, tosca.Value{}, receiver1)
+				// perform static call and forward the result
+				code0 = append(code0, []byte{
+					byte(vm.STATICCALL),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				code1 := pushCallArguments(call, sufficientGas, tosca.Value{}, receiver2)
+				// perform call and return whether it was successful
+				code1 = append(code1, []byte{
+					byte(call.callType),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				// perform sstore (should revert in static mode)
+				code2 := []byte{
+					byte(vm.PUSH1), byte(0),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.SSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}
+
+				state := WorldState{
+					sender0:   Account{},
+					receiver0: Account{Code: code0},
+					receiver1: Account{Code: code1},
+					receiver2: Account{Code: code2},
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender0,
+					Recipient: &receiver0,
+					GasLimit:  sufficientGas,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Errorf("execution was not successful or failed with error %v", err)
+				}
+				if !slices.Equal(result.Output, bytes.Repeat([]byte{0}, 32)) {
+					t.Errorf("%s did not enforce static mode and returned %v", callName, result.Output)
+				}
```

Comment:
Aren't we missing a check here for the reverted storage mentioned above?

Reply:
I don't want to enforce a certain storage state in a failed case, this is handled by the interpreter not the processor.